<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Matthew Stephens" />

<meta name="date" content="2016-01-22" />

<title>Intro to Pott Data</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<div class="container-fluid main-container">

<!-- tabsets -->
<script src="libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">hgen 48600</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/stephens999/hgen48600">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Intro to Pott Data</h1>
<h4 class="author"><em>Matthew Stephens</em></h4>
<h4 class="date"><em>2016-01-22</em></h4>

</div>

<div id="TOC">
<ul>
<li><a href="#read-data">Read Data</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2017-01-05</p>
<p><strong>Code version:</strong> 8937f3c0e1d6b49a79a773e6940c3d92a272cbe1</p>
<p>This is a very brief intro to some data from Sebastian Pott (a research associate here), which he kindly agreed to share with us for teaching purposes. The initial goal is to fit a Markov chain to some of these data, and compare it to a model where the data are independent. (I suspect that neither of these models will be a good fit to the data, although I have not actually tried it yet.)</p>
<div id="read-data" class="section level2">
<h2>Read Data</h2>
<p>The data related to methylation or non-methylation at positions along the genome. However, these are not standard methylation data - essentially the experimental protocol is such that locations where the DNA is more “open” are more likely to read out as “methylated”. So think of the methylation/non-methlation as a noisy measure of “open”/“closed”.</p>
<p>Sebastian processed the sequence data, aligning reads to the genome, to produce a bed file <code>Test_region_NOM37_methylation_C_in_GpC_hg38_chr9_131946471_134832670.bed</code>. Columns 1-6 contain the position of the nucleotide in the genome. Column 7 shows the the number of reads with a methylated C in this position. Column 8 contains the total number of reads mapping to the correct strand at this position. So Column 7 should be less than Column 8, and the difference in the two columns is the number of unmethylated reads at that position.</p>
<pre class="r"><code>x = read.table(&quot;../data/potts/Test_region_NOM37_methylation_C_in_GpC_hg38_chr9_131946471_134832670.bed&quot;,stringsAsFactors = FALSE, header=FALSE)
head(x)</code></pre>
<pre><code>    V1        V2        V3      V4 V5 V6 V7 V8
1 chr9 131946692 131946693 region1  0  +  0  3
2 chr9 131946703 131946704 region1  0  +  0  3
3 chr9 131946722 131946723 region1  0  +  2  3
4 chr9 131946726 131946727 region1  0  +  0  1
5 chr9 131946749 131946750 region1  0  +  0  1
6 chr9 131946796 131946797 region1  0  +  0  2</code></pre>
<pre class="r"><code>#check that, as claimed, V7 is never larger than V8
sum(x$V7&gt;x$V8)</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>#make tables of coverage and counts
table(x$V7)</code></pre>
<pre><code>
    0     1     2     3     4     5     6     7     8     9    10    11 
62784 11464  1550   345    99    56    30     8     8     4     3     1 
   12    14    21 
    1     1     1 </code></pre>
<pre class="r"><code>table(x$V8)</code></pre>
<pre><code>
    1     2     3     4     5     6     7     8     9    10    11    12 
27285 18698 11944  7585  4652  2736  1633   857   469   249   122    59 
   13    14    15    17    20    21 
   36    21     5     2     1     1 </code></pre>
<p>The coverage (column 8) is pretty low, and for now - mostly for simplicity - we are going to ignore differences in coverage. So we’re just going to ignore column 8. And, again for simplicity, we are going to binarize column 7 (most observations are 0 or 1 anyway!). So here goes:</p>
<pre class="r"><code>m = 1*(x$V7&gt;0)
head(m,100)</code></pre>
<pre><code>  [1] 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 [36] 0 0 0 0 0 0 1 0 1 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 1 0
 [71] 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 1 1 1 0 1 0 0 0 0 0 0 1</code></pre>
<p>OK, so now <code>m</code> represents a very noisy binary observation of whether each position is “open” or not. The exercise here is to fit and compare likelihoods for models where i) elements of <span class="math inline">\(m\)</span> are independent; b) <span class="math inline">\(m\)</span> follows a Markov chain. (It would also be good form to do some more simple plots or summaries to familiarize yourself more with the data before fitting the models!)</p>
<ol style="list-style-type: lower-alpha">
<li>Bernoulli model
<ol style="list-style-type: lower-roman">
<li>Fit a Bernoulli model to the data. That is, assuming the elements <span class="math inline">\(m_i\)</span> are independent Bernoulli(<span class="math inline">\(q\)</span>) for some <span class="math inline">\(q\)</span>, estimate <span class="math inline">\(q\)</span> by maximum likelihood.<br />
</li>
<li>If <span class="math inline">\(\hat{q}\)</span> denotes the maximum likelihood estimate, and <span class="math inline">\(l_{bern}\)</span> denotes the log-likelihood function, compute the value of <span class="math inline">\(l_{bern}(\hat{q}; m)\)</span>.</li>
</ol></li>
<li>Markov Chain
<ol style="list-style-type: lower-roman">
<li><p>Fit a 2-state (0/1) Markov Chain to the vector <span class="math inline">\(m\)</span> by maximum likelihood. Here the parameter is the 2 by 2 transition matrix <span class="math inline">\(P\)</span> between the two states 0 and 1. [To get the Markov chain started, assume that the first observation <span class="math inline">\(m_1\)</span> is Bernoulli($), where <span class="math inline">\(\hat{q}\)</span> is the parameter estimated above.]</p></li>
<li><p>Again, compute the log-likelihood value at the mle you have found. That is, if <span class="math inline">\(l_{markov}\)</span> denotes the log-likelihood function and <span class="math inline">\(\hat{P}\)</span> the mle for <span class="math inline">\(P\)</span>, compute <span class="math inline">\(l_{markov}(\hat{P}; m)\)</span>.</p></li>
</ol></li>
<li>Compare: comment on how different the two log-likelihoods you have computed are.</li>
</ol>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.1 (2016-06-21)
Platform: x86_64-apple-darwin11.0.0 (64-bit)
Running under: OS X 10.12.2 (Sierra)

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] knitr_1.14

loaded via a namespace (and not attached):
 [1] magrittr_1.5    formatR_1.4     tools_3.3.1     htmltools_0.3.5
 [5] yaml_2.1.13     Rcpp_0.12.8     stringi_1.1.1   rmarkdown_1.0  
 [9] stringr_1.0.0   digest_0.6.10   evaluate_0.9   </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
